src/Cli/dotnet/commands/dotnet-complete/Suggest.cs

```diff
@@ -11,56 +15,69 @@ namespace Microsoft.DotNet.Cli
 {
     internal static class Suggest
     {
-        public static IEnumerable<string> TargetFrameworksFromProjectFile()
+        public static SuggestDelegate TargetFrameworksFromProjectFile()
         {
-            try
+            return (ParseResult parseResult, string textToMatch) =>
             {
-                return GetMSBuildProject()?.GetTargetFrameworks().Select(tf => tf.GetShortFolderName()) ?? Empty<string>();
-            }
-            catch (Exception)
-            {
-                return Empty<string>();
-            }
+                try
+                {
+                    return GetMSBuildProject()?.GetTargetFrameworks().Select(tf => tf.GetShortFolderName()) ?? Empty<string>();
```

Comment:
Is this expected to throw an exception under any conditions, or is this in a try/catch block just in case?  Would like to ensure that we don't expect to throw if the project doesn't exist, or under other conditions that would be considered normal.

Reply:
Usually `catch (Exception)` will hide bugs. Although for auto completion there is no way to surface the error anyway (not even print to screen), and there is nothing the user can take action on, catch everything is fine here.

Reply:
Is there a chance `GetShortFolderName()` would throw? `Select` gets evaluated lazily so any exceptions thrown inside of it would be evaluated outside of these try-catch blocks).

Reply:
> Is there a chance GetShortFolderName() would throw?

I don't think so

Reply:
Cool.  Just wanted to make sure that the try/catch isn't hiding an expected exception path, since exceptions are more costly.